33-lifecycle-lab_v16/src/03-custominput.jsx

```diff
@@ -1,10 +1,80 @@
-/**
- * Implment an input component that shows 4
- * input boxes writing user input in the currently
- * focused panel
- *
- * See lab description on the webpage for a live example
- * how this should work.
- */
-
-
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class SquareFocus extends React.Component{
+
+    constructor(props) {
+        super(props);
+        this.state = {texts : []};
+    }
+
+
+    setFocus = (el,index) =>{
+        if(el!=null){
+            el.focus()
+            el.style.backgroundColor = "#ccc";

```

Comment:
Still some things worth mentioning: Don't set the background colour for a focused element in JS. Write the CSS rule `div:focus { ... }`